flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkManifestUtil.java

```diff
@@ -73,21 +78,43 @@ static ManifestOutputFileFactory createOutputFileFactory(
         attemptNumber);
   }
 
+  /**
+   * Write the {@link WriteResult} to temporary manifest files.
+   *
+   * @param result all those DataFiles/DeleteFiles in this WriteResult should be written with same
+   *     partition spec
+   */
   static DeltaManifests writeCompletedFiles(
-      WriteResult result, Supplier<OutputFile> outputFileSupplier, PartitionSpec spec)
+      WriteResult result,
+      Supplier<OutputFile> outputFileSupplier,
+      Map<Integer, PartitionSpec> specsById)
       throws IOException {
 
     ManifestFile dataManifest = null;
     ManifestFile deleteManifest = null;
+    PartitionSpec spec = null;
```

Comment:
Updated the code implementation. Now get the true partition spec from the data files/delete files.

Reply:
> We use a SerializableTable instance to create IcebergStreamWriter. The PartitionSpec of SerializableTable is fixed and will not change after the job started. While the PartitionSpec for IcebergFilesCommitter is refreshed with the table snapshot changing. This could fail the fink sink job when updating the partition spec in another job. Because we use the wrong partition spec to write those DataFiles/DeleteFiles to ManifestFile.

your original description analyzed the problem very well. I am wondering if we should just pass the same read-only `SerializableTable` to `IcebergFilesCommitter` so that it also use the same table spec as the `IcebergStreamingWriter/RowDataTaskWriterFactory`. 

@hililiwei @Reo-LEI what do you think?

Reply:
There is a problem that the job's partition spec maybe not same as the one in the state when we restore the state from the restart job.

Reply:
I am only talking about write path. 

for read path, you are passing in the `specsById` which should work.

Reply:
Sorry, I got mixed up with something. Keep the original implementation.

>  I am wondering if we should just pass the same read-only SerializableTable to IcebergFilesCommitter so that it also use the same table spec as the IcebergStreamingWriter/RowDataTaskWriterFactory.

I think we should not use the `SerializableTable`, it can not commit the new files change. Or I missed something.

Reply:
sorry for the confusion. For commit, we still need the TableLoader to load a regular table at the committer operator initialization. I was saying that we pass a read-only table or partition spec to the committer for writing manifest file. this partition spec is always the same as the writer operator.

Reply:
@stevenzwu Thanks for the explanation, keep the original implementation and pass the same partition spec to `IcebergFilesCommitter `